qiskit/compiler/transpile.py

```diff
@@ -234,10 +234,14 @@ def _parse_basis_gates(basis_gates, backend, circuits):
     if basis_gates is None or (isinstance(basis_gates, list) and
                                all(isinstance(i, str) for i in basis_gates)):
         basis_gates = [basis_gates] * len(circuits)
-    # no basis means don't unroll (all circuit gates are valid basis)
-    basis_gates = [[inst.name for inst, _, _ in circuit.data] if basis is None
-                   else basis for basis, circuit in zip(basis_gates, circuits)]
 
+    # no basis means don't unroll (all circuit gates are valid basis)
+    for index, circuit in enumerate(circuits):
+        basis = basis_gates[index]
```

Comment:
it's cleaner to just iterate over the zip, rather than index first.

Reply:
The oneliner that was there was very hard for me to read... Since I have to udpate the Nones in in the `basis_gates` list, I dont see how zip could help...